tests/certification/pubsub/azure/eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,301 @@
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// Run dummy app
+		Step(app.Run(dummyAppID, fmt.Sprintf(":%d", appPort+portOffset*10),
+			dummyApp(dummyAppID))).
+		// Run the dummy Dapr sidecar with the dummyconsumer component, to be used for publishing messages to a topic with no subscribers (unused)
+		Step(sidecar.Run(sidecarDummyName,
+			embedded.WithComponentsPath("./components/dummyconsumer"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*10),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*10),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*10),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*10),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+		Step("publish messages to topic1", publishMessages(nil, sidecarName1, topicName1, consumerGroup1, consumerGroup2)).
+		// Start a sidecar with a component refering eh connection string of unusedTopic
+		Step("publish messages to unUsedTopic", publishMessages(nil, sidecarDummyName, unUsedTopic)).
+		Step("verify if app1 has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup1)).
+		Step("verify if app2 has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup2)).
+		Step("reset", flow.Reset(consumerGroup2)).
+
+		// Test : multiple publisher with different partitionkey, multiple subscriber with same consumer ID
+		// Run subscriberApplication app3
+		Step(app.Run(appID3, fmt.Sprintf(":%d", appPort+portOffset*2),
+			subscriberApplication(appID3, topicName1, consumerGroup2))).
+
+		// Run the Dapr sidecar with the component 3.
+		Step(sidecar.Run(sidecarName3,
+			embedded.WithComponentsPath("./components/consumer3"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*2),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*2),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*2),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*2),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// publish message in topic1 from two publisher apps, however there are two subscriber apps (app2,app3) with same consumerID
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName1, topicName1, consumerGroup2)).
+		Step("publish messages to topic1", publishMessages(metadata1, sidecarName2, topicName1, consumerGroup2)).
+		Step("verify if app2, app3 together have recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup2)).
+
+		// Test : Entitymanagement
+		// Run subscriberApplication app4
+		Step(app.Run(appID4, fmt.Sprintf(":%d", appPort+portOffset*3),
+			subscriberApplication(appID4, topicToBeCreated, consumerGroup4))).
```

Comment:
when is this topic deleted? if this test needs to test entity creation every time. otherwise it will not be creating that event hub but reuse the existing event hub created when the first test was run ...

Reply:
Thanks for the catch, this has been fixed as part of cleanup step up in the test.